lib/Backend/Lower.cpp

```diff
@@ -8824,7 +8851,16 @@ Lowerer::LowerLdArrViewElem(IR::Instr * instr)
         }
         done = instr;
     }
-    InsertMove(dst, src1, done);
+    if (dst->IsInt64())
```

Comment:
IMO it would be nice if InsertMove took care of this rather than doing the check at all the call sites (same with ChangeToAssign)